noir/compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -75,11 +75,12 @@ impl GeneratedAcir {
         std::mem::take(&mut self.opcodes)
     }
 
-    /// Updates the witness index counter and returns
-    /// the next witness index.
-    pub(crate) fn next_witness_index(&mut self) -> Witness {
+    /// Returns the current witness index and updates it 
+    /// so that we have a fresh witness the next time this method is called
+    pub(crate) fn get_current_witness_and_update(&mut self) -> Witness {
+        let next_witness_index = Witness(self.current_witness_index);
         self.current_witness_index += 1;
-        Witness(self.current_witness_index)
+        next_witness_index
```

Comment:
Hah wow. I've clearly spent no time looking at rust bc I thought this was an error. Can't believe this is the return syntax in rust.